modules/util/source-formatter/src/main/resources/dependencies/replacements.json

```diff
@@ -21,6 +21,11 @@
 		],
 		"to": "getFriendlyURL(new InfoItemReference(param#0#, param#1#), param#2#)"
 	},
+	{
+		"from": "LayoutFriendlyURLComposite(Layout paramName, String paramName)",
+		"issueKey": "LPD-8025",
+		"to": "LayoutFriendlyURLComposite(param#0#, param#1#, false)"
+	},
```

Comment:
For constructor patterns, we used the attribute `"skipParametersValidation": true` shouldn't this still be the case?

It's common to fill the constructor on new variable declarations with cases that will trigger a message without skipping the validation. 
Previously, it was more valuable to make the switch directly by only checking the quantity without checking the types, which is what happens when you add this attribute.
This is the way this case has been dealt with before, it is open to evolution because it has an edge case if it is possible for a class to have two public constructors with the same number of parameters.

Sorry if I went on too long, the point is: we have to maintain consistency, will another developer looking at it know whether or not to use the attribute in the case of constructures? 
We have a few options: 
Verify that checking the type won't break previous cases (there's even a PR with more than 600 cases for constructors to enter from a break-change that occurred in the master). Or we can align your PR to work like the old standard.
Or even follow the second option, but in another ticket, do it the old way and do the improvement, analysis, and refactoring in another ticket.

Please let me know your thoughts about it. Cc @drewbrokke

Reply:
Also, we created the `ci:test:hotel` suite case to run our tests to avoid spending CI resources on suites like relevant and stable.

Our tests never will run on Liferay code so they never will break either of these suites. Aligned with 
Shuyang in the past we agreed that only `ci:test:sf` and `ci:test:hotel` were enough for the type of tests.  
If you change any of SF's base classes, utilities, javaParser, etc. Then you need to run these more complete suites of cases and ask for Alan Huang review, for example. If not, the review doesn't need to be escalated.
@kyle-miho  @drewbrokke

Reply:
@nicolas-sss I wasn't aware that skipParameterValidation is required to be added to constructors as I did not do a replace for a Constructor yet, and it is also not in the confluence tuttorial either. I dont get why that is the standard though, wouldnt we want to validate constructor parameters? Because wouldnt it have the same logic as non constructors, where if we only care about checking the paramNums and not the paramTypes as well, we could be upgrading some incorrect edge cases? I dont understand why the "old standard" is skipping validation.

Regardless I will re-submit the PR with the added parameter then, but I think its something we should clarify, if there is a  reason for it that makes sense it should be added to the confluence doc as well to prevent confusion. But if its to help with the validation I dont understand why.

Reply:
For example if we are only updating certain constructor methods like (string, string) and not (long, string) we will do incorrect upgrades, same as a  method call. Im not sure why constructor replacements are being treated differently, when in essence they should be the same unless if I am missing something

Reply:
Hi, @kyle-miho when implemented it was noticed that constructor cases for more than 90% of the cases filled the parameters with cases that will trigger messages, so constructor cases will never do the replacement. 
Like: 
A: `new LayoutFriendlyURLComposite(var.getLayout(), var.getUrl());` way more common than 

B: `new LayoutFriendlyURLComposite(layout, friendlyURL)`

So to bring more value it was decided (based on the cases found until the moment) that we should skip parameter type validations on constructor cases. Yeah, I'm aware of the possibility of this edge case, and because of that, I have suggested the possibility of improving this case. But basically only add the way more manual steps on our side than possible incorrect replacements that may never occur.

Let's say that a client code will have 100 constructor cases with 1 incorrect replacement being possible to happen. But for these 100, 95 of them directly pass the values on the parameters like the example A.

This will lead to the case that the SF will only do 5 cases if we don't skip the parameter check, and the dev will have to do it 95 times manually.